js/shopping-list.js

```diff
@@ -0,0 +1,44 @@
+// let Item = require('../js/shopping-list-item.js');
+
+class List {
+  constructor() {
+    this.items = [];
+
+  }
+  addItem(item) {
+    if (item instanceof Item) {
+      this.items.push(item);
+    } else {
+      throw new Error;
+    }
+  }
+  removeItem(item) {
+    if (item instanceof Item) {
+      let index = this.items.indexOf(item);
+      if (index >= 0) {
+        this.items.splice(index, 1);
+      } else {
+        throw new Error;
+      }
+    } else
+      if (item === undefined) {
+        this.items.pop();
+      } else {
+        throw new Error;
+      }
+  }
+
+  render() {
+    let start = '<ul>';
```

Comment:
Again string template literals would be a good things to use here rather than adding string together.